numpydoc/validate.py

```diff
@@ -125,8 +127,12 @@ def extract_ignore_validation_comments(filepath: os.PathLike) -> Dict[int, List[
     dict[int, list[str]]
         Mapping of line number to a list of checks to ignore.
     """
-    with open(filepath) as file:
```

Comment:
I would tweak this slightly.

```python
numpydoc_ignore_comments = {}

try:
    with open(filepath) as file:
        last_declaration = 1
        declarations = ["def", "class"]
        for token in tokenize.generate_tokens(file.readline):
            if token.type == tokenize.NAME and token.string in declarations:
                last_declaration = token.start[0]
            if token.type == tokenize.COMMENT:
                match = re.match(IGNORE_COMMENT_PATTERN, token.string)
                if match:
                    rules = match.group(1).split(",")
                    numpydoc_ignore_comments[last_declaration] = rules
except (OSError, TypeError):  # can be None, nonexistent, or unreadable
    pass

return numpydoc_ignore_comments
```

Reply:
I actually prefer the current approach as I always try to limit the scope of `try/except` as much as possible. With your code there is a single return path, which is nice. However the scoping for the `try/except` is broader, which means you risk accidentally catching a TypeError somewhere in the remaining 10 lines that do more than just "try to open the file". And I don't think the interpretation of the current code is too bad: "try to open the file and if you can't, return an empty set. if you can, proceed to parse it"